tidy3d/components/mode/mode_solver.py

```diff
@@ -0,0 +1,2205 @@
+"""Solve for modes in a 2D cross-sectional plane in a simulation, assuming translational
+invariance along a given propagation axis.
+"""
```

Comment:
I'm confused about this file as github is showing it as entirely new, but it seems pretty much the same, are there any other changes except supporting `ModeSimulation`?

Reply:
Yeah, it's almost exactly the same file. The only changes are the addition of the validator `_validate_mode_simulation` which ensures consistency between the ModeSimulation and the ModeSolver, the addition of ModeSimulation to MODE_SIMULATION_TYPE and ModeSimulationData to MODE_SIMULATION_DATA_TYPE, and the function `sim_data` is modified if simulation is a ModeSimulation